server/data/courseClient.ts

```diff
@@ -19,9 +19,15 @@ export default class CourseClient {
     return (await this.restClient.get({ path: paths.courses.show({ id }) })) as Course
   }
 
-  async findOfferings(courseId: Course['id']): Promise<Array<CourseOffering>> {
+  async findOfferings(id: Course['id']): Promise<Array<CourseOffering>> {
```

Comment:
is it because it matches the path variable name? We could rename that to `courseId` if we think it'd be clearer

Reply:
For a method looking for a single resource like `find` an `id` parameter is clear enough as there's only one thing and you'd infer it from the service/client.

Because this method is looking for a differently-named resource to its client (the `offering` rather than `course`), I think it's a bit more complicated, but I'd tend to assume here that the `id` we're passing in could be the `offeringId` based on the method's name (although it's further complicated because it's plural...). If we were to call this method `findOfferingsByCourseId`, then `id` would be clear enough, so I'd maybe say let's do one or the other (rename method or variable).

Reply:
Re: re: renaming the param, I'm not sure exactly what the conventions are, but I know in previous projects we've gone for the following style of routes

`courses/:id` - is fine because there's one entity
`courses/:courseId/offerings/:offeringId` - is a bit clearer than `courses/:id/offerings/:offeringId`, particularly in controller actions where you need to fetch things by different IDs.

Although with the `static-path` thing we haven't done this as we were building up from the original path that only had one resource so `id` was fine! I don't think it's a big deal here yet though, but maybe something to consider in future.

Reply:
I've had a look around and I think it might be good to rename the params actually? In a separate PR though